app/controllers/application_controller.rb

```diff
@@ -10,11 +10,24 @@ class ApplicationController < ActionController::Base
   helper LinksHelper
   helper_method :current_user
   helper_method :sso_identity
+  helper_method :current_estate
 
   def current_user
     sso_identity&.user
   end
 
+  def current_estate
+    return unless sso_identity
+    @_current_estate ||= begin
+      estate_id = session[:current_estate]
```

Comment:
This could result in a 404 which would be very confusing to the user and when debugging if the current_estate is removed from the user sso settings and then the user logs in again.